### PR TITLE
Tab Router Initial Feedback

### PR DESCRIPTION
**Pull request by ZakRaymer**

### CODE REVIEW

File: src/examples/routes/Page2b.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+import { Text, TouchableOpacity, View } from 'react-native';
+import { compose, withHandlers } from 'recompose';
+
+const Page2b = props => (
```
Comment (jottenlips) [inline src/examples/routes/Page2b.tsx]: I generally avoid numbers and shortened spelling in variable names

File: src/examples/store.ts [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { routerReducer } from '../index';
 
 const combinedReducers = combineReducers({
     router: routerReducer({
+        activeTab: 1,
```
Comment (jottenlips) [inline src/examples/store.ts]: we should start on the furthest left view, we may even want to make this configurable, 0

File: src/components/TabRouter.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        </TouchableOpacity>
+      ))}
+    </View>
+  );
+
+  const tabView = (
+    <View style={{ height: `Calc(100vh - ${props.viewHeightReduction}px` }}>
+      {props.tabViews[props.activeTabIndex]}
+    </View>
+  );
```
Comment (jottenlips) [inline src/components/TabRouter.tsx]: Can we give tabs a selected state so I know which tab I am on?

File: src/routerReducer.ts [hunk 4]
```diff
@@ -66,7 +72,8 @@ const routerReducer = (config: {
                 ...state,
                 isNavigating: true,
                 history,
-                destinations: [...state.destinations, payload]
+                destinations: [...state.destinations, payload],
+                tabRoutes: state.tabRoutes.map((route, index) => index === state.activeTab ? tabRoute : route )
```
Comment (jottenlips) [inline src/routerReducer.ts]:
```suggestion
                tabRoutes: state.tabRoutes.map((route, index) => (index === state.activeTab) ? tabRoute : route )
``` 
helps my eyes a bit

File: src/routerReducer.ts [hunk 5]
```diff
@@ -93,6 +100,7 @@ const routerReducer = (config: {
         [RESET_NAVIGATION]: (state) => ({
             ...state,
             history: [state.history[0], last(state.history)],
+            tabRoutes: state.tabRoutes.map((route, index) => index === state.activeTab ? [route[0], last(route)] : route),
```
Comment (jottenlips) [inline src/routerReducer.ts]: same as above

File: src/examples/routes/Page2.tsx [hunk 6]
```diff
@@ -39,8 +39,7 @@ export default compose(
     withHandlers({
         nextPage: props => _e => {
             props.setRoute({
-                route: 'Page3',
-                navigationTitle: "Page 3"
```
Comment (jottenlips) [inline src/examples/routes/Page2.tsx]: would like to keep an example somewhere of a title set on navigation

**Comment by jottenlips** [comment]
Works great! just a few things

File: src/examples/store.ts (hunk 2 above)
Comment (benminer) [reply src/examples/store.ts]: Definitely need it configureable. Maybe pass down an initialTab prop?

File: src/routerReducer.ts [hunk 7]
```diff
@@ -102,10 +110,15 @@ const routerReducer = (config: {
         }),
 
         [NAVIGATE_BACK_COMPLETE]: (state) => {
-            const history = [...state.history];
+            const history = [...state.tabRoutes[state.activeTab]];
```
Comment (benminer) [inline src/routerReducer.ts]: nice

File: src/components/TabRouter.tsx (hunk 3 above)
Comment (benminer) [reply src/components/TabRouter.tsx]: ^^. Maybe even an active and inactive color prop for each? That will definitely be a thing Colton wants.

**Comment by jottenlips** [comment]
When this gets merged I will import it into the new monorepo and archive this repository

File: src/examples/store.ts (hunk 2 above)
Comment (ZakRaymer) [reply src/examples/store.ts]: This is the pass in example. It currently works that way.

**Comment by jottenlips** [comment]
we are checking for backwards compatability

**Comment by jottenlips** [comment]
all good!

**Pull request merged**